### PR TITLE
Clarify lightweight policy requirements

### PR DESCRIPTION
**Pull request by aleksei-fedotov**
Lightweight policy imposes `noexcept` requirement on a function call `operator()` of its node's body.
This patch adds missing description and updates corresponding example.

Notify: @alexey-katranov @vlserov 

Signed-off-by: Fedotov, Aleksei <aleksei.fedotov@intel.com>

### CODE REVIEW

File: source/elements/oneTBB/source/flow_graph/functional_node_policies.rst [hunk 1]
```diff
@@ -56,6 +56,10 @@ with the behavior defined by the ``lightweight`` policy. For example, if the def
 ``Policy`` is ``queueing``, specifying ``lightweight`` as the ``Policy`` value is equivalent to
 specifying ``queueing_lightweight``.
 
+The ``lightweight`` policy requires a function call ``operator()`` of a node's
+body be ``noexcept``. If it is not the case, specifying ``lightweight`` policy
+to the node has no effect.
+
```
Comment (akukanov) [inline source/elements/oneTBB/source/flow_graph/functional_node_policies.rst:62]: It's true not just ``lightweight`` but also for combined policies, right? I would rephrase to something like "The function call operator of a node body must be ``noexcept`` for lightweight policies to have effect".

Also I think the "hint" nature of the lightweight policies should be better described in the specification; "helps to reduce the overhead" is very vague. This improved general description would also mention that the effect is implementation-specific and "best effort", completely ignoring the lightweight hint is valid. Then we can describe ``noexcept`` as the necessary condition (like I did above) without additionally saying that there is no effect if the condition is not met (because "no effect" is always a valid implementation option).

File: source/elements/oneTBB/source/flow_graph/functional_node_policies.rst [hunk 2]
```diff
@@ -48,14 +48,18 @@ of a predecessor to handle this.
 Lightweight
 -----------
 
-This policy helps to reduce the overhead associated with the execution scheduling of the node.
+This policy suggests that the node body takes little time to process, as a non-binging hint for
+an implementation to reduce overheads associated with the node execution. Any applied optimization
+must have no observable side effects on the node and graph execution.
```
[the remark this answers is not in the document]
Comment (akukanov) [reply source/elements/oneTBB/source/flow_graph/functional_node_policies.rst:53]: As written, it specifies absence of observable side effects **from the optimizations**, not from execution. E.g. you cannot simply skip executing the body :)
As it was not quite clear to you, could you maybe suggest changes that would make it more clear?

[the remark this answers is not in the document]
Comment (aleksei-fedotov) [reply source/elements/oneTBB/source/flow_graph/functional_node_policies.rst:53]: I think the confusion came from the fact that the optimization is performed by the library and it targets execution overheads only, not optimizing anything in the user function. And I read it from the perspective of the graph user - a node implementer, not the implementer of the library specification.

**Review by aleksei-fedotov — COMMENTED** [review]
Now it is much clearer! Thanks!
Sorry, I cannot approve because the system does not allow me to do that - I am the PR owner or by some other reason.
